x/nexus/exported/types.go

```diff
@@ -374,6 +382,10 @@ func (m WasmMessage) ValidateBasic() error {
 		return fmt.Errorf("invalid wasm message payload hash")
 	}
 
+	if len(m.SourceTxID) == 0 {
```

Comment:
The EVM core gateway requires tx id to be exactly 32 bytes. Looks like nexus gateway just encodes arbitrary length bytes for it (it'll work for hex hash but not for others potentially). Let's use all 0's there if it can't be converted